### ISSUE TITLE
Ray Tracing with file

### ISSUE THREAD

**Issue by brycemines** [mines-oceanography/mantaray#20]
Once the PR for CartesianFile is merged:

- [x] create a new pr for ray tracing with depth
- [x] have the ray tracing function use the Cartesian file to access the depth and calculate group velocity.
- [x] update the k derivatives and add function to calculate depth gradient
